### PR TITLE
Rewrite tax module

### PR DESCRIPTION
**Pull request by ashu555**
1. Check Tax Creation from Admin Panel
2. All the dependecies related to it in cart, checkout payment etc in both client and Admin panel.
3. Check Invoice creation order execution in detail
4. Coupon code discounts.

 Run following migrations:
php artisan migrate --path="database/migrations/2020_08_13_045954_add_processing_fee_to_invoices.php"
php artisan migrate --path="database/migrations/2020_08_18_122436_add_cin_to_settings_table.php"

### CODE REVIEW

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Invoice issue still not fixed in this PR please check and let me know.
Please can we have a demo session for tax PR? So that i can start testing.

**Comment by SarfrazS** [comment]
migration command is wrong that's why the order is not generating now we correct the command
php artisan migrate --path="database/migrations/2020_08_13_045954_add_processing_fee_to_invoices.php"

File: app/Http/Controllers/Front/CheckoutController.php [hunk 1]
```diff
@@ -102,14 +106,13 @@ public function checkoutForm(Request $request)
         }
         if (\Session::has('items')) {
             $content = \Session::get('items');
-            $attributes = $this->getAttributes($content);
+            $taxConditions = $this->getAttributes($content);
         } else {
             $content = Cart::getContent();
-            $attributes = $this->getAttributes($content);
+            $taxConditions = $this->getAttributes($content);
```
Comment (mverma16) [inline app/Http/Controllers/Front/CheckoutController.php:112]: else is not required, remove duplicate statements

File: app/Http/helpers.php [hunk 2]
```diff
@@ -221,3 +233,179 @@ function getIndianCurrencyFormat($number)
         return $thecash;
     }
 }
+
+function bifurcateTax($taxName, $taxValue, $currency, $state, $price)
```
Comment (mverma16) [inline app/Http/helpers.php:237]: write proper tests for significant methods like `bifurcateTax`

**Review by mverma16 — CHANGES_REQUESTED** [review]

**Review by SarfrazS — COMMENTED** [review]
Please unable to install and not able to use existing DB getting some error.
Please can u connect and resolve my issue so that I can continue testing.
image
image

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Checking  all PR's in single branch enhancement-razorpay-plugin issues are there updated on below doc. 
Still i am checking. It is taking time to creating db and calculation. 
https://docs.google.com/document/d/1HqdRicmgmGPnqVsS_AJhe4tiqufVzYAYENVT4E8JpFg/edit

File: app/Http/Controllers/Front/CheckoutController.php (hunk 1 above)
Comment (ashu555) [reply app/Http/Controllers/Front/CheckoutController.php:112]: Fixed

File: app/Http/helpers.php (hunk 2 above)
Comment (ashu555) [reply app/Http/helpers.php:237]: Fixed

**Comment by mverma16** [comment]
Do not see any changes since I last reviewed, have you pushed your changes?

**Review by SarfrazS — APPROVED** [review]
working fine, i am approving this PR some unknown issues are there pointed to the developer he working on it.
https://docs.google.com/document/d/1HqdRicmgmGPnqVsS_AJhe4tiqufVzYAYENVT4E8JpFg/edit

**Pull request closed without merging**